algosdk/future/transaction.py

```diff
@@ -82,20 +82,22 @@ def get_txid(self):
         txid = base64.b32encode(txid).decode()
         return encoding._undo_padding(txid)
 
-    def sign(self, private_key):
+    def sign(self, private_key, skip_rekey_check=False):
```

Comment:
Do we expect the user to know when to use it?

Reply:
I expect the user to ignore it, which is why it's defaulted to `False`. The function comment notes that `True` is `typically used for size estimation`. I could change it to do a separate, private `_sign_and_skip_rekey_check` instead, so it's not exposed to the user.

Reply:
If the user doesn't need to know about it. Let's make it private.